packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -307,30 +318,67 @@ export abstract class BaseServiceV2<
   public async run(): Promise<void> {
     this.done = false
 
-    // Start the metrics server if not yet running.
-    if (!this.metricsServer) {
-      this.logger.info('starting metrics server')
+    // Start the app server if not yet running.
+    if (!this.server) {
+      this.logger.info('starting app server')
+
+      // Start building the app.
+      const app = express()
+
+      // Body parsing.
+      app.use(bodyParser.json())
+      app.use(bodyParser.urlencoded({ extended: true }))
+
+      // Logging.
+      app.use(
+        morgan((tokens, req, res) => {
+          return [
+            tokens.method(req, res),
+            tokens.url(req, res),
+            tokens.status(req, res),
+            JSON.stringify(req.body),
+            '\n',
+            tokens.res(req, res, 'content-length'),
+            '-',
+            tokens['response-time'](req, res),
+            'ms',
+          ].join(' ')
+        })
+      )
 
-      await new Promise((resolve) => {
-        const app = express()
+      // Metrics.
+      // Will expose a /metrics endpoint by default.
+      app.use(
+        promBundle({
+          promRegistry: this.metricsRegistry,
+          includeMethod: true,
+          includePath: true,
+          includeStatusCode: true,
+        })
+      )
 
-        app.get('/metrics', async (_, res) => {
-          res.status(200).send(await this.metricsRegistry.metrics())
+      // Health status.
+      app.get('/healthz', async (req, res) => {
+        return res.json({
+          ok: this.healthy,
```

Comment:
We should add a script to package.json called `yarn generate-sha` or something.   That script should run git rev-parse HEAD or whatever that command is to get the current git sha and generate a file called something like gitsha.txt.   We should then fs.readFile() to get the git sha and return it here so we not only know the service is up but can easily verify what version is running.

Reply:
Adding in a separate PR